test/y2network/config_writer/sysconfig_dns_test.rb

```diff
@@ -0,0 +1,163 @@
+      allow(Yast::Execute).to receive(:on_target!)
+    end
+
+    context "when dhcp_hostname is changed to true" do
+      let(:dhcp_hostname) { true }
+
+      it "sets DHCLIENT_SET_HOSTNAME to 'yes'" do
+        expect(Yast::SCR).to receive(:Write)
+          .with(Yast::Path.new(".sysconfig.network.dhcp.DHCLIENT_SET_HOSTNAME"), "yes")
+        writer.write(dns, old_dns)
+      end
+    end
+
+    context "when dhcp_hostname is changed to false" do
+      let(:dhcp_hostname) { false }
+
+      it "writes DHCLIENT_SET_HOSTNAME to 'no'" do
+        expect(Yast::SCR).to receive(:Write)
+          .with(Yast::Path.new(".sysconfig.network.dhcp.DHCLIENT_SET_HOSTNAME"), "no")
+        writer.write(dns, old_dns)
+      end
+    end
+
+    context "when dhcp_hostname is not changed" do
+      let(:old_dhcp_hostname) { dhcp_hostname }
+
+      it "does not write DHCLIENT_SET_HOSTNAME" do
+        expect(Yast::SCR).to_not receive(:Write)
+          .with(Yast::Path.new(".sysconfig.network.dhcp.DHCLIENT_SET_HOSTNAME"), anything)
+        writer.write(dns, old_dns)
+      end
+    end
+
+    context "when dhcp_hostname is nil" do
+      let(:dhcp_hostname) { nil }
+
+      it "does not write DHCLIENT_SET_HOSTNAME" do
+        expect(Yast::SCR).to_not receive(:Write)
+          .with(Yast::Path.new(".sysconfig.network.dhcp.DHCLIENT_SET_HOSTNAME"), anything)
+        writer.write(dns, old_dns)
+      end
+    end
+
+    it "updates the hostname" do
+      expect(Yast::Execute).to receive(:on_target!).with("/bin/hostname", "myhost")
+      expect(Yast::SCR).to receive(:Write)
+        .with(Yast::Path.new(".target.string"), "/etc/hostname", "myhost.example.net\n")
+      writer.write(dns, old_dns)
+    end
+
+    context "when sendmail update script is installed" do
+      before do
+        allow(Yast::FileUtils).to receive(:Exists).with(/sendmail/).and_return(true)
+      end
+
+      it "updates MTA configuration" do
+        expect(Yast::Execute).to receive(:on_target!).with(/sendmail/)
+        writer.write(dns, old_dns)
+      end
+    end
+
+    context "when sendmail update script is not installed" do
+      before do
+        allow(Yast::FileUtils).to receive(:Exists).with(/sendmail/).and_return(false)
+      end
+
+      it "updates MTA configuration" do
+        expect(Yast::Execute).to_not receive(:on_target!).with(/sendmail/)
+        writer.write(dns, old_dns)
+      end
+    end
+
+    it "updates DNS policy" do
+      expect(Yast::SCR).to receive(:Write)
+        .with(Yast::Path.new(".sysconfig.network.config.NETCONFIG_DNS_POLICY"), "auto")
+      writer.write(dns, old_dns)
+    end
+
+    it "updates the list of search domains" do
+      expect(Yast::SCR).to receive(:Write)
+        .with(
+          Yast::Path.new(".sysconfig.network.config.NETCONFIG_DNS_STATIC_SEARCHLIST"),
+          "example.net example.org"
+        )
+      writer.write(dns, old_dns)
+    end
+
+    it "updates the list of name servers" do
```

Comment:
```suggestion
    it "updates the list of nameservers" do
```

Reply:
It is true that the variable is called `nameserver`, but we use `name server` in the UI, so I will keep it here as it is (against you have a strong opinion against).